openbb_platform/providers/binance/openbb_binance/models/crypto_historical.py

```diff
@@ -0,0 +1,98 @@
+"""Binance Crypto Historical WS Data."""
+
+import json
+import logging
+from datetime import datetime, timedelta
+from typing import Any, AsyncGenerator, AsyncIterator, Dict, Literal, Optional
+
+import websockets
+from openbb_core.provider.abstract.fetcher import Fetcher
+from openbb_core.provider.standard_models.crypto_historical import (
+    CryptoHistoricalData,
+    CryptoHistoricalQueryParams,
+)
+from pydantic import Field
+
+# pylint: disable=unused-argument, arguments-differ
+
+logger = logging.getLogger(__name__)
+
+
+class BinanceCryptoHistoricalQueryParams(CryptoHistoricalQueryParams):
+    """Binance Crypto Historical Query Params."""
+
+    tld: Optional[Literal["us", "com"]] = Field(
+        default="us", description="Top-level domain of the Binance endpoint."
+    )
+    lifetime: Optional[int] = Field(
+        default=60, description="Lifetime of WebSocket in seconds."
+    )
+
+
+class BinanceCryptoHistoricalData(CryptoHistoricalData):
+    """Binance Crypto Historical Data."""
+
+    __alias_dict__ = {
+        "symbol": "s",
+        "close": "c",
+        "open": "o",
+        "high": "h",
+        "low": "l",
+        "volume": "v",
+    }
+    event_type: Optional[str] = Field(
+        default=None,
+        description="Event type",
+        alias="e",
+    )
+    quote_asset_volume: Optional[str] = Field(
+        default=None,
+        description="Total traded quote asset volume",
+        alias="q",
+    )
+
+
+class BinanceCryptoHistoricalFetcher(Fetcher):
```

Comment:
I think this model name and Fetcher class is deceiving. It is not at all like the other provider/standard models with this name.  This is not historical data, and the pattern itself is significantly different. It needs to be differentiated. 
Like how you have added an additional `atransform` method, there should be something like `wsconnect` that accepts a URL and **kwargs where the core logic can be reused.


There should be a WebSocket standard model which ships with the connection logic that others can